appcues/src/main/java/com/appcues/debugger/ui/DebuggerComposition.kt

```diff
@@ -70,6 +71,15 @@ internal fun DebuggerComposition(viewModel: DebuggerViewModel, onDismiss: () ->
             debuggerState = debuggerState,
             debuggerViewModel = viewModel,
         )
+
+        val capture = debuggerState.screenCapture.value
+        if (capture != null) {
```

Comment:
is this the best way to conditionally show this dialog based on state?

Reply:
found an interesting guide from androiddevelopers that showcase the showing of a modal.

https://developer.android.com/codelabs/basic-android-kotlin-compose-viewmodel-and-state#8

it seems like we would use a StateFlow and collectAsState to check its value. the way you have now is probably working tho

Reply:
Interesting, I think this has some similarities already to what we are doing in the DebuggerComposition LaunchedUIStateEffect, where the VM UIState has `collectAsState` and then this dialog shows on `Expanded(ScreenCapture)` state (causing this recomposition to get the `debuggerState.screenCapture` here and check if not null)